src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,107 @@
+/*
+    Jon Kaimmer
+    CS410 Software Engineering
+    Spring 2023
+    Junit Testing Assignment
+
+    Spring Manipulation
+ */
+
+import java.lang.IllegalArgumentException;
+import java.lang.IndexOutOfBoundsException;
+import java.lang.NullPointerException;
+import java.util.*;
+
+
 public class StringManipulation implements StringManipulationInterface {
 
+    public String s;
+
     @Override
     public String getString() {
-        return null;
+        return s;
     }
 
     @Override
     public void setString(String string) {
+        s = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if (s == null || s.isEmpty()){
+            return 0;
+        }
+        String[] words = s.split("\\s+");
+        return words.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+
+        if(s == null){
+            throw new NullPointerException("string cannot be null");
+        }
+        if (n <= 0) {
+            throw new IllegalArgumentException("n must be greater than zero.");
+        }
+        if (n > s.length()){
+            throw new IndexOutOfBoundsException("n is greater than the string length.");
+        }
+
+        StringBuilder sb = new StringBuilder();
```

Comment:
Here you can instantiate StringBuilder() by passing an existing string to it. After, you can call StringBuilder.remove(int index) to remove characters at a given position. If maintainSpacing == true, StringBuilder.setCharAt(int index, char ch) can be called to replace chars with ' '.